### ISSUE TITLE
Trailing slashes removed from registered routes

### ISSUE THREAD

**Issue by kashike** [pippo-java/pippo#420]
```java
@POST("/foo/bar/")
public void foo() {
    // ...
}
```

```
[00:20:41] [XNIO-1 task-1/DEBUG] [DefaultRouter]: Add route for POST '/foo/bar'
[00:20:41] [XNIO-1 task-1/DEBUG] [PippoFilter]: Request POST '/foo/bar/'
```

https://github.com/pippo-java/pippo/issues/366#issuecomment-381921555

**Comment by decebals** [comment]
I think that the problem is in [ControllerRegistry](https://github.com/pippo-java/pippo/blob/master/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java#L202) where the "/" from the end of path (if it exists) is removed.

**Comment by kashike** [comment]
It would be [here](https://github.com/pippo-java/pippo/blob/master/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java#L217-L220) instead, no?

**Comment by decebals** [comment]
I am very busy these days. I think that is easy to fix this issue (in few words, just remove `StringUtils.removeEnd(path, "/")` and make some tests to see that it works).

**Comment by mhagnumdw** [comment]
I think I debugged the method [ControllerRegistry.createControllerRoutes](https://github.com/pippo-java/pippo/blob/release-1.8.0/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java#L181) on all flows.

If a route "/hello/" should be registered as "/hello/" (with / to the end), them I think, just like you, that just removing `StringUtils.removeEnd(path, "/")` is the solution.

Maybe there another scenario that I did not think.

I can implement some tests. Whould the tests be for [ControllerRegistry](https://github.com/pippo-java/pippo/blob/release-1.8.0/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java) (ControllerRegistryTest)?

**If so, in summary, the tests would be:** Create some Controllers with and without `@Path`, with some methods with and without `@GET`, also including `@GET` without value. Then check if the routes created have the expected path. **Right?**

**Comment by decebals** [comment]
@mhagnumdw 
The test described by you is perfect.
I prefer to create a unit test (not to use pippo-test) if it's possible (something similar with `DefaultRouterTest`).

**Comment by mhagnumdw** [comment]
@decebals 
I am writing the tests.

I noticied that the methods [ControllerRegistry.register(Package...)](https://github.com/pippo-java/pippo/blob/release-1.8.0/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java#L69) and [ControllerRegistry.register(Class<? extends Controller>...)](https://github.com/pippo-java/pippo/blob/release-1.8.0/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java#L101) produce different results.

**For example:**

**Assuming there is only the WithoutPathController class inside the package.**

```java
controllerRegistry.register(WithoutPathController.class); // build two routes
```

```java
controllerRegistry.register(WithoutPathController.class.getPackage()); // do not build route
```

**What is the correct behavior?**

```java
public static class WithoutPathController extends Controller {

    @GET
    public void example0() {
    }

    @GET("/example1")
    public void example1() {
    }

    // without http method
    public void withoutHttpMethod() {
    }

}
```

**Comment by decebals** [comment]
@mhagnumdw 
I think that it's a bug. In both situations we must have two routes.

**Comment by decebals** [comment]
The bug is fixed by #423. I released a new SNAPSHOT and if you confirm that the issue is resolved, I will release a new version. Thanks!

**Comment by mhagnumdw** [comment]
@decebals 
I'm doing some testing on my application and I'm having problems. Some routes have apparently not been registered. I'll post more details soon.

@kashike 
If possible, can you also do some testing?

**Comment by mhagnumdw** [comment]
@decebals 
Actually everything is working so well that I thought it was a bug :smile: 

I was trying to access the method `loadPage` below from the url /pg/p/cofreCons (without / to the end) and of course I would not access :)

```java
@Path("/pg/p/cofreCons")
public class CofreConsultaController extends Controller {
    @GET("/")
    public void loadPage() {
        getResponse().render("/pg/p/cofreCons");
    }
}
```

**Comment by decebals** [comment]
@mhagnumdw 
I released a new version (1.9.0). 
Can you update the [changelog](https://github.com/pippo-java/pippo/blob/master/CHANGELOG.md) using the information from https://github.com/pippo-java/pippo/compare/release-1.8.0...HEAD (put only the important commits). Thanks!

**Comment by mhagnumdw** [comment]
@decebals 
Done.
